lib/Conversion/ArcToLLVM/LowerArcToLLVM.cpp

```diff
@@ -173,6 +173,73 @@ struct StorageGetOpLowering : public OpConversionPattern<arc::StorageGetOp> {
   }
 };
 
+struct AssertOpLowering : public OpConversionPattern<arc::AssertOp> {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(arc::AssertOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+
+    rewriter.replaceOpWithNewOp<scf::IfOp>(
+        op, adaptor.getProperty(),
+        [&](auto &builder, auto loc) {
+          // if property is true then assert will not execute
+          builder.template create<scf::YieldOp>(loc);
```

Comment:
Why not negate the property and only have a then-branch?

Reply:
I implemented it like this at first.
to negate the condition I need to `xor` it with `true`. 
It generates more operations, but maybe in some following step, the `then` branch will be removed, by some `CFG` simplifications. 

```
bool cond = 0;
assert(0);
--> 
bool cond = 0;
bool trueVal = 1;
bool invertedCond = cond ^ trueVal;
if(invertedCond) {}
--------------
bool cond = 0;
if(cond) {// do nothing}
else {}
```
I think I should negate it as you suggested, the code looks naive, and If it doesn't the xor is faster than the branch predictor anyway. I just wanted to share my thoughts.